sdk/storage/Azure.Storage.Blobs/src/BlobClientSideEncryptor.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Cryptography;
+using Azure.Storage.Cryptography;
+using Azure.Storage.Cryptography.Models;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
+
+namespace Azure.Storage.Blobs
+{
+    internal class BlobClientSideEncryptor
+    {
+        private readonly IKeyEncryptionKey _keyEncryptionKey;
+        private readonly string _keyWrapAlgorithm;
+
+        public BlobClientSideEncryptor(ClientSideEncryptionOptions options)
+        {
+            _keyEncryptionKey = options.KeyEncryptionKey;
+            _keyWrapAlgorithm = options.KeyWrapAlgorithm;
+        }
+
+        /// <summary>
+        /// Applies client-side encryption to the data for upload.
+        /// </summary>
+        /// <param name="content">
+        /// Content to encrypt.
+        /// </param>
+        /// <param name="metadata">
+        /// Metadata to add encryption metadata to.
+        /// </param>
+        /// <param name="async">
+        /// Whether to perform this operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Cancellation token.
+        /// </param>
+        /// <returns>Transformed content stream and metadata.</returns>
+        public async Task<(Stream, Metadata)> ClientSideEncryptInternal(
+            Stream content,
+            Metadata metadata,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            if (_keyEncryptionKey == default || _keyWrapAlgorithm == default)
+            {
+                throw Errors.ClientSideEncryption.MissingRequiredEncryptionResources(nameof(_keyEncryptionKey), nameof(_keyWrapAlgorithm));
+            }
+
+            //long originalLength = content.Length;
+
+            (Stream nonSeekableCiphertext, EncryptionData encryptionData) = await ClientSideEncryptor.EncryptInternal(
+                content,
+                _keyEncryptionKey,
+                _keyWrapAlgorithm,
+                async,
+                cancellationToken).ConfigureAwait(false);
+
+            //Stream seekableCiphertext = new RollingBufferStream(
+            //        nonSeekableCiphertext,
+            //        EncryptionConstants.DefaultRollingBufferSize,
+            //        GetExpectedCryptoStreamLength(originalLength));
```

Comment:
Please cleanup commented out code here and there. Either we should work items for these improvements or implement them right away.

Reply:
This was old code that got moved around in the refactors. Is obsolete and will be removed.